### PR TITLE
Travel method buttons components

### PR DESCRIPTION
**Pull request by Amutha37**
Please review the component for travelMethodButtons in  `components/TravelMethodButtons/TravelMethodButtons.jsx`
Trello link :  https://trello.com/c/z50v3GrA/297-dev-travel-method-selection-page

New commit 7

- Import the 'modeOfTransport' and 'transportIcons' directly travelMethodButtons component.

- There no reason to call it as a props .

### CODE REVIEW

File: components/TravelMethodButtons/TravelMethodButtons.jsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import { Box, Button, Text, SimpleGrid, Icon, Flex } from '@chakra-ui/react';
+
+export function TravelMethodButtons({
```
Comment (hqtan) [inline components/TravelMethodButtons/TravelMethodButtons.jsx:3]: can we rename this function as `TravelMethodButtonsContainer` please? :pray: 

```suggestion
export function TravelMethodButtonsContainer({
```

File: components/TravelMethodButtons/TravelMethodButtons.jsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  transportIcon,
+  modesOfTransport,
+  handleTransportMode,
+  transportMode,
+}) {
+  return (
+    <SimpleGrid
+      columns={3}
+      defaultValue={transportMode}
```
Comment (hqtan) [inline components/TravelMethodButtons/TravelMethodButtons.jsx:12]: Why is there a need for a 'default' value for transport method?
Unless you have a good reason why we need one, I suggest removing it.

```suggestion
```

File: components/TravelMethodButtons/TravelMethodButtons.jsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
+import { Box, Button, Text, SimpleGrid, Icon, Flex } from '@chakra-ui/react';
+
+export function TravelMethodButtons({
+  transportIcon,
+  modesOfTransport,
+  handleTransportMode,
+  transportMode,
```
Comment (hqtan) [inline components/TravelMethodButtons/TravelMethodButtons.jsx:7]: do we need to pass in these values as parameters to this function?
- `transportIcon` - this is a list of transport icons from `utils/constants.js` - why can't we `import` the icons here instead?
-  `modesOfTransport` - this is a list of accepted transport methods from `utils/constants.js`. Why can't we `import` this directly?
- `handleTransportMode` - this is the click event handler for the buttons. I think it makes sense to define the `TravelMethodButtons` button click event handler within the `TravelMethodButtonsContainer` itself... but there are pros for passing it in as a parameter too. What do you think @batteries76 , @OlaShabalina, @ymloh ?
- `transportMode` - this is a default value for a travel method, Which we don't really need. Under what circumstances would a survey offer 'default' answers for questions? Does it apply to the survey we are building?

```suggestion
```

File: components/TravelMethodButtons/TravelMethodButtons.jsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+              top={['3px', '2px']}
+              pb={['4px', '3px']}
+              width={['30px', '32px']}
+              height={['30px', '32px']}
+              mb="12px"
+            >
+              <Icon as={transportIcon[i]} />
+            </Box>
+            {mode}
+          </Button>
```
Comment (hqtan) [inline components/TravelMethodButtons/TravelMethodButtons.jsx:45]: can we extract this Button out into its own component, and call the component `TravelMethodButton` ?
`TravelMethodButton` can be extracted out into its own file (preferably), or left in this file.

**Review by hqtan — CHANGES_REQUESTED** [review]
Hi @Amutha37 ,

I've left some suggestions for your consideration. :bow: 

Another request: can we ensure we are consistent with our use of double-quotes for strings? Thx!

File: components/TravelMethodButtons/TravelMethodButtons.jsx (hunk 3 above)
Comment (Amutha37) [reply components/TravelMethodButtons/TravelMethodButtons.jsx]: hgtan question > transportIcon - this is a list of transport icons from utils/constants.js - why can't we import the icons here instead?
modesOfTransport - this is a list of accepted transport methods from utils/constants.js. Why can't we import this directly?

Answer > These icons are used in array form together with the text are displayed with the array of name listed in constant.jsx. This code to retrieve icons sit well with the map method used for retrieving the button text.
I hope this answer your question.

question > handleTransportMode - this is the click event handler for the buttons. I think it makes sense to define the TravelMethodButtons button click event handler within the TravelMethodButtonsContainer itself... but there are pros for passing it in as a parameter too. What do you think @batteries76 , @OlaShabalina, @ymloh ?

Answer > Currently the I'm focussing only on design. I haven't change the functionality.

Question > transportMode - this is a default value for a travel method, Which we don't really need. Under what circumstances would a survey offer 'default' answers for questions? Does it apply to the survey we are building?

Answer > This has been removed now.

File: components/TravelMethodButtons/TravelMethodButtons.jsx (hunk 1 above)
Comment (Amutha37) [reply components/TravelMethodButtons/TravelMethodButtons.jsx]: This is resolved

File: components/TravelMethodButtons/TravelMethodButtons.jsx (hunk 2 above)
Comment (Amutha37) [reply components/TravelMethodButtons/TravelMethodButtons.jsx]: This line is removed.

File: components/TravelMethodButtons/TravelMethodButtons.jsx (hunk 3 above)
Comment (hqtan) [reply components/TravelMethodButtons/TravelMethodButtons.jsx]: Hi @Amutha37,

I think you've misunderstood my reasoning for removing `transportIcon` and `transportMode` as parameter values for the `TravelMethodButtonsContainer` function. It has little to do with using `map` method.

just to be clear, this is the change I proposed:
image

my reasoning:
- parameters are 'variable' inputs - for values that would be used to calculate an output
- in the case of a `TravelMethodButtonsContainer` component, these travel method icons are the reason why this component exists; if you were able to pass in fruit icons instead, would it still be a `TravelMethodButtonsContainer` component?
- a similar argument can be made for `transportMode`

Functionally speaking `TravelMethodButtonsContainer` component will work the same way whether you decide to pass in `transportIcon` and `transportMode` as parameter values or with an `import` statement. It has more to do with the reasons 'why' one uses function parameters to obtain values, versus using imports to obtain values instead..

Comment (OlaShabalina) [reply components/TravelMethodButtons/TravelMethodButtons.jsx]: I believe the advantage of having a tree structure for components is the possibility of making changes to each part of the page without affecting the page. However, on a side note, if we have a page which is working - we can always accept this solution for now and re-factor the code later one once the rest is done.

**Comment by Amutha37** [comment]
I will remove transport Icon as props. It doesn’t parent component doesn’t
need this. I will use import  them inside the button component .

File: components/TravelMethodButtons/TravelMethodButtons.jsx (hunk 3 above)
Comment (Amutha37) [reply components/TravelMethodButtons/TravelMethodButtons.jsx]: The above question has been addressed . As at last commit. These 2 props doesn't make sense as a props so I have use them directly inside the button container.

File: components/TravelMethodButtons/TravelMethodButtons.jsx (hunk 4 above)
Comment (Amutha37) [reply components/TravelMethodButtons/TravelMethodButtons.jsx]: This will be addressed in a new PR.

**Comment by hqtan** [comment]
Hi @Amutha37 ,

I'm going to close this PR, in favour of #70 .
I've compared both PRs, and came to the conclusion that all the significant changes in this PR are in #70 .

Please let me know if I've erred in my judgement and many apologies in advance. We can always re-open this PR.

**Comment by amu3** [comment]
yes you can actually. By doing this the oder of PR review is not necessary
.

**Pull request closed without merging**